docs/proposals/scheduling/binding-priority-preemption/README.md

```diff
@@ -0,0 +1,286 @@
+
+authors:
+- "@whitewindmills"
+
+reviewers:
+- "@RainbowMango"
+- "@XiShanYongYe-Chang"
+- "@chaunceyjiang"
+
+approvers:
+- "@RainbowMango"
+
+creation-date: 2024-05-27
+
+notes: This proposal refers to ResourceBinding and ClusterResourceBinding collectively as binding.
+---
+
+# Binding Priority and Preemption
+
+## Summary
+
+Currently, the scheduler only schedules workloads in order, that is, the workload triggered first can be scheduled first.
+And when a new binding has certain scheduling requirements that makes it infeasible on any member cluster,
+it may stay in the scheduling queue, until sufficient resources are free, and it can be scheduled.
+
+This proposal proposes the concept of priorities and preemption for bindings in Karmada and how the priority impacts scheduling and preemption of bindings
+when the member cluster runs out of resources.
+- The priority is independent of its workload type and must be one of the valid values, there is a total order on the values.
+- Preemption is the action taken when an important binding requires resources or conditions which are not available in all member clusters. So, one or more bindings need to be preempted to make room for the important binding.
+
+## Motivation
+
+It is fairly common in large-scale clusters to have more tasks than what Karmada can handle, so scheduling responses can become slow.
+Often times the workload is a mix of high priority critical tasks, and non-urgent tasks that can wait.
+Karmada should be able to distinguish these workloads in order to decide which ones should be scheduled sooner, which ones can wait and which ones can be preempted.
+
+### Goals
+
+- How to specify priority and preemption strategy for bindings.
+- Define how the order of these priorities are specified.
+- Define how new priority levels are added.
+- Effect of priority on scheduling and preemption.
+- Define the concept of binding preemption in Karmada.
+- Define scenarios under which a binding may get preempted.
+- Define mechanics of preemption.
+
+### Non-Goals
+
+- How preemption works in spread constraints.
+
+## Proposal
+
+### Terminology
+
+When a new binding has certain scheduling requirements that makes it infeasible on any member cluster,
+scheduler may choose to clear the scheduling result of lower priority bindings to satisfy the scheduling requirements of the new binding.
+We call this operation "binding preemption". Binding preemption is distinguished from "[policy preemption](../policy-preemption/README.md)" where high-priority policies preempt low-priority policies.
+
+### User Stories (Optional)
+
+#### As a user, I want to the task I submitted to be scheduled as soon as possible.
+
+I submit a workload that is prioritized above other workloads, but do not wish to discard existing work by preempting scheduled bindings.
+I hope that the high priority workload will be scheduled ahead of other queued bindings, as soon as sufficient cluster resources "naturally" become free.
```

Comment:
```suggestion
I usually classify my workloads based on priority and want high-priority workloads to be scheduled ahead of others, rather than waiting to be scheduled in the order they were queued. This will allow me to maintain the quality of service for important workloads even during resource contention.
```